engine/src/w32spec.cpp

```diff
@@ -455,33 +455,52 @@ char *MCS_get_canonical_path(const char *path)
 {
 	char *t_path = NULL;
 	char *t_curdir = NULL;
+	char *t_livecode_path = NULL;
+	char *t_livecode_path_start;
 
 	if (path == NULL)
 		return NULL;
+	
+	// SN-2015-09-04: [[ Bug 15814 ]] Backslashes are not valid filename chars
+	//  on Windows, so we can accept Windows paths using backslashes (for the 
+	//  people who are still using them in LiveCode scripts).
+	bool t_use_backslashes;
+	t_use_backslashes = false;
+	uint4 i = 0;
+	while (path[i] != NULL && !t_use_backslashes)
+		t_use_backslashes = path[i++] == '\\';
+	
+	t_livecode_path = strclone(path);
+	t_livecode_path_start = t_livecode_path;
 
-	if (path[0] == '/' && path[1] != '/')
+	if (t_use_backslashes)
```

Comment:
This doesn't quite work though. If there is a backslash in the path you are mapping / -> \ and \ -> / -- *then* checking for absoluteness. However consider the following path:
```
//foo/bar/baz\foobar.txt
```

This would fail to be recognized as a absolute path as you would be testing:
```
\\foo\bar\baz/foobar.txt
```

Whilst it is possible for NTFS filenames to contain \, I suspect bad things happen in the UI on Windows and elsewhere if such a thing is encountered. Given this is the case, I wonder if the best thing to do is to not do any \ <-> mapping here, and instead check for either / or \ in the appropriate places.

This will preserve the user's file path, whilst still prepending the CWD if it is a relative path.

It is then up to the FileSystem implementation functions to map appropriately - which I believe they already do.